crates/sparrow-materialize/src/materialize/materialization_control.rs

```diff
@@ -0,0 +1,221 @@
+use std::sync::Arc;
+
+use error_stack::{IntoReport, ResultExt};
+use futures::{StreamExt, TryStreamExt};
+use sparrow_api::kaskada::v1alpha::ProgressInformation;
+use sparrow_runtime::s3::S3Helper;
+use tokio::task::JoinHandle;
+
+use crate::{Error, Materialization};
+
+/// The control for a materialization process.
+///
+/// This handles communication with the materialization process,
+/// allowing a user to manipulate the process via the service layer.
+pub struct MaterializationControl {
+    /// Unique identifier of the materialization
+    pub id: String,
```

Comment:
Does this need the ID? I wonder if we could keep the ID a purely service concept:

1. Materialization: How we actually do materialization
2. Materialization Control: How we manage an asynchronous materialization
3. Service: How we implement the gRPC interface, including the map from IDs to materializations.

In that model, it seems like the ID doesn't matter here.

Reply:
Hm, we could. But I currently have it for logging within the control API. I'll leave it for debugging purposes, but added a note to remove it if decide it's unnecessary